core.py

```diff
@@ -69,6 +72,64 @@ def analyze_file(_file, result, settings):
         entropy = settings.entropy if settings.entropy else HIGH_ENTROPY_EDGE
+        regexlist = settings.regex_list if settings.regex_list else REGEXP_LIST
+        singleregex = settings.single_regex if settings.single_regex else SINGLE_REGEX
+
+        if settings.regexlist:
+            # have to read line by line instead of words
+            try:
+                file = open(f'{_file}')
+                lines = file.readlines()
+                foundings = []
+                found = 0    
+                file = yaml.safe_load(open(f'{regexlist}'))
+                regexArray = file.get("signatures")
+                for _key in lines:
+                    for _finding in regexArray:
+                        if any('regex' in _key for _key in _finding):
+                            regex = f"{_finding['regex']}"               
+                            pattern = re.compile(regex)
+                            found = pattern.findall(_key)
+                            if (found == ['']):
+                                print(f'Possibe match: {colored(_finding["name"], "cyan")}')
+                                print(f'File: {colored(_file, "green")}')
+                                print(f'Key: {colored(_key, "magenta")}')
+                                foundings.append(f'Possibly matched: {colored(_finding["name"], "cyan")} with key: {colored(_key, "magenta")}')
+                                data = {"Finding" : f'{_finding["name"]}',  'Key': f'{_key}', 'File':f'{_file}'}
+                                result.put(data)          
+                            if (found != [] and found != ['']) :
+                                i = len(found);
+                                for var in list(range(i)):
+                                    print(f'Possibe match: {colored(_finding["name"], "cyan")}')
+                                    print(f'File: {colored(_file, "green")}')
+                                    print(f'Key: {colored(found[var] + "", "magenta")}\n')
+                                    foundings.append(f'Possibly matched: {colored(_finding["name"], "cyan")} with key: {colored(_key, "magenta")}')
+                                    data = {"Finding" : f'{_finding["name"]}',  'Key': f'{found[var]}', 'File':f'{_file}'}
+                                    result.put(data)
+                                       
+            except Exception as e:
+                logger.error("while trying to open " + str(_file) + ". Details:\n" + str(e))
+
+        if settings.singleregex:
+            #Single regex availible now, multiple inline regexes will be addressed in future updates
+            try:
+                file = open(f'{_file}')
+                lines = file.readlines()
+                # foundings = []
+                found = 0
+                with open(_file) as f:
+                    for _key in lines:
+                        pattern = re.compile(singleregex)
```

Comment:
it doesn't search through extensions and filenames and in regex.yaml there are many rules against it. It would be nice to add a check against the filename too